### PR TITLE
fix some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
This PR removes a few needless clones and adds `# Safety` sections to docs of unsafe functions.

### CODE REVIEW

File: cranelift-codegen/src/isa/mod.rs [hunk 1]
```diff
@@ -22,7 +22,6 @@
 //! ```
 //! # extern crate cranelift_codegen;
 //! # #[macro_use] extern crate target_lexicon;
```
Comment (abrown) [inline cranelift-codegen/src/isa/mod.rs]: Should we remove these two commented-out lines as well?

File: cranelift-entity/src/boxed_slice.rs [hunk 2]
```diff
@@ -27,6 +27,10 @@ where
 {
     /// Create a new slice from a raw pointer. A safer way to create slices is
     /// to use `PrimaryMap::into_boxed_slice()`.
+    ///
+    /// # Safety
+    ///
+    /// relies on `raw` pointing to a valid slice of `V`s
```
Comment (abrown) [inline cranelift-entity/src/boxed_slice.rs]: Before @bnjbvr gets to this: he'll say this should be capitalized and punctuated.

File: cranelift-codegen/src/isa/mod.rs (hunk 1 above)
Comment (llogiq) [reply cranelift-codegen/src/isa/mod.rs]: Those are not commented out, they are just marked as invisible on rustdoc render.

**Comment by llogiq** [comment]
Added the punctuation and made it a complete sentence.

**Comment by llogiq** [comment]
Added the punctuation and made it a complete sentence.

File: cranelift-codegen/src/isa/mod.rs (hunk 1 above)
Comment (abrown) [reply cranelift-codegen/src/isa/mod.rs]: Cool, didn't know about that, 👍

**Pull request merged**